### PR TITLE
Video input api

### PR DESCRIPTION
**Pull request by JeroMiya**
Bindings for the video input part of the display config. Also updated the OSVR-Core snapshot to the version that includes the new API.

### CODE REVIEW

File: ClientKit/Display.cs [hunk 1]
```diff
@@ -26,7 +26,9 @@
 using ViewerCount = System.UInt32;
 using EyeCount = System.Byte;
 using SurfaceCount = System.UInt32;
-using DistortionPriority = System.Int32;
+using DistortionPriority = System.Int32;
+using DisplayInputCount = System.Byte;
```
Comment (rpavlik) [inline ClientKit/Display.cs:30]: am I correct in remembering that .NET doesn't distinguish signed vs. unsigned for int8s?

File: ClientKit/Display.cs [hunk 2]
```diff
@@ -66,6 +68,26 @@ public override string ToString()
...
+    /// Dimensions for a display input.
+    /// </summary>
+    public struct DisplayDimensions
+    {
+        public DisplayDimension Width { get; set; }
+        public DisplayDimension Height { get; set; }
+    }
+
+    /// <summary>
+    /// Projection clipping planes for a give viewer-eye-surface locus.
```
Comment (rpavlik) [inline ClientKit/Display.cs]: Typo "give"->"given", and probably should be annotated that the only function that returns them returns the planes' locations at unit distance, so you'll usually want to divide by your "near" distance (but at that point you might as well just let us do the matrices for you...)

File: ClientKit/Display.cs [hunk 3]
```diff
@@ -368,6 +439,31 @@ public Viewport GetRelativeViewportForViewerEyeSurface(ViewerCount viewer, EyeCo
...
         }
 
+        /// <summary>
+        /// Get the index of the display input for a surface seen by an eye of a
+        /// viewer in a display config.
+        /// 
+        /// This is the OSVR-assigned display input: it may not (and in practice,
+        /// usually will not) match any platform-specific display indices. This function
+        /// exists to associate surfaces with video inputs as enumerated by
+        /// osvrClientGetNumDisplayInputs().
```
Comment (rpavlik) [inline ClientKit/Display.cs]: perhaps refer to `GetNumDisplayInputs()` here instead?

**Comment by rpavlik** [comment]
Just a few (primarily comment-related) nitpicks, looks good other than that, so merge when you're happy with my notes.

File: ClientKit/Display.cs (hunk 1 above)
Comment (JeroMiya) [inline ClientKit/Display.cs:30]: .Net has an unsigned `byte` (unsigned int8) and a signed `sbyte` (signed int8), which are C# keyword aliases for the `System.Byte` and `System.SByte` types, respectively. I think it's Java that doesn't have a built-in unsigned 8-bit primitive.

**Pull request merged**